### ISSUE TITLE
advanced support for template strings

### ISSUE THREAD

**Issue by InSyncWithFoo** [astral-sh/ty#528]
[PEP 750](https://peps.python.org/pep-0750/) introduces <i>template strings</i>, also known as t-strings:

```python
template: Template = t'Hello {name}'
```

According to the PEP, this feature is expected to be used for "safety checks, web templating, domain-specific languages, and more".

---

Obviously, this is a big feature and needs some design, but here are a few ideas to get things started:

```python
type_checker = 'ty'                   # Literal['ty']
template = t'{type_checker} is fast'  # LiteralTemplate[t'{type_checker} is fast', type_checker: Literal['ty']]

reveal_type(template.strings)         # tuple[Literal[''], 'is fast']
reveal_type(template.interpolations)  # tuple[LiteralInterpolation[Literal['ty']]]
```

**Comment by AlexWaygood** [comment]
See also the discussion at https://github.com/python/cpython/issues/133970.

I'm not sure I'd want to introduce any kind of dedicated `Type` variant or special form for template strings until I'm a bit clearer on what the use cases are, and/or whether there's going to be any attempt to standardise support for generic `Template`s in typeshed. It's generally good to keep special-casing (and complexity in our model) to a minimum.

**Comment by carljm** [comment]
We do now have basic support for template strings in 3.14, but it's just what the typeshed annotations give; there's no support for a "template literal" type that actually knows its details.
